updatecli/updatecli.d/windows-2022-images.yml

```diff
@@ -0,0 +1,48 @@
+---
+name: Bump azure `windows 2022` image version
+
+scms:
+  default:
+    kind: github
+    spec:
+      user: "{{ .github.user }}"
+      email: "{{ .github.email }}"
+      owner: "{{ .github.owner }}"
+      repository: "{{ .github.repository }}"
+      token: "{{ requiredEnv .github.token }}"
+      username: "{{ .github.username }}"
+      branch: "{{ .github.branch }}"
+
+sources:
+  lastReleaseVersion:
+    kind: shell
+    name: Get the latest `windows 2022` image version
+    spec:
+      # Reuse environment variables from parent shell
+      environments:
+        - name: PATH # If `az` is in a non-usual installation path
+        - name: AZURE_CLIENT_ID
+        - name: AZURE_CLIENT_SECRET
+        - name: AZURE_TENANT_ID
+      command: az login --service-principal -u $AZURE_CLIENT_ID -p $AZURE_CLIENT_SECRET -t $AZURE_TENANT_ID && az vm image list --location eastus --publisher MicrosoftWindowsServer --offer WindowsServer --sku 2022-datacenter-core-g2 --all --query "[?offer=='WindowsServer'].version" -o tsv | sort -u | tail -n 1
+
+targets:
+  updateVersion:
+    name: Update azure `windows 2022` image version in locals
+    sourceid: lastReleaseVersion
+    kind: file # not terraform/file as per updatecli/updatecli#1859 need to use matchpattern and replacepattern for now
+    scmid: default
+    spec:
+      file: "locals.pkr.hcl"
+      matchpattern: (windows_image_version.*(\r\n|\r|\n).*(\r\n|\r|\n).*\"2022\".*=.*\")(.*)(\".*)
+      replacepattern: ${1}{{ source "lastReleaseVersion" }}"
```

Comment:
Fails with the following error:

```
"update pipeline: yaml: line 42: did not find expected key"
```

Reply:
WDYT about specifying the windows versions in a YAML file (which is easy to parse with updatecli) and use the Terraform functions to read it as a local:

```yaml
# packer-versions.yaml
---
windows_image_version:
  2019: "17763.6293.240905"
  2022: "20348.2762.241006"
  "22.04": "latest" #just to avoid error on run for linux, but not used
```

and 

```
locals {
  # ...
   windows_image_version = yamldecode(file("${path.module}/packer-versions.yaml")).windows_image_version
  # ...
}
```